content/english/python-discord-translate/creating-discord-bot/writing-the-bot/_index.md

```diff
@@ -0,0 +1,154 @@
+
+**Event (@bot.event)**: Decorators like `@bot.event` tell the library “run this function when a specific event happens” (here: when the bot is ready).
+
+**Slash commands (`@bot.tree.command`)**: These create modern commands that appear when users type `/` in Discord. They’re registered with Discord’s servers and may take a short time to sync.
+
+**Interaction**: Represents a user using a slash command. `interaction.user` gives you the person, and `interaction.response.send_message(...)` sends a reply.
+
+**Bot vs Client**: `commands.Bot` adds helpful command features (like command trees) on top of the lower-level Discord client.
+
+**Intents**: Permissions telling Discord what data you want (messages, members, etc.). Using fewer keeps your bot simpler and safer.
+
+**Exceptions (try / except)**: The `try:` block runs code that might fail; `except Exception as e:` catches errors so your bot can log them instead of crashing.
+
+</details>
+
+{{< alert theme="danger" >}}<img src="../../media/NF_mascot.jpg" alt="Nuvi mascot" width="70" style="float:right;margin:0 0 6px 10px;" />
+<strong>Beware!</strong> Putting your BOT_TOKEN directly in the code is not advised in normal circumstances, this is just for workshop simplification. Normally should use environment variables or a secure storage location.{{< /alert >}}
+
+---
+
+## 3. Run the Bot
+1. Open `bot.py` in the editor.
+1. Open the Run & Debug view (play icon with a bug).
+1. If asked to select a configuration, choose `Python File`.
+1. Click the green Run/Debug button.
+1. Watch the Debug Console / Terminal for:
+```
+Logged in as <your bot name>
+Synced X application command(s).
+
+```
+![Output in terminal](../../media/running.png)
+
+---
+
+## 4. Test the /hello Command in Discord
+1. Invite your bot to a server where you have permissions (you can generate an invite link from the Developer Portal under "OAuth2 → URL Generator" with the `bot` and `applications.commands` scopes, as well as the `Send Messages` text permission).
+![Invite bot with correct scopes](../../media/urlgen.png)
+![Bot invite link](../../media/bot-perms.png)
+![Bot in member list](../../media/url.png).
+1. Copy the url and paste it into your browser. That should open a Discord page asking you to select a server to invite the bot to. Select your test server.
+![Invite bot to server](../../media/allow.png)
+1. Make sure the bot is running, you should see it online in the member list.
+![Bot online in member list](../../media/bot-online.png)
+1. In any text channel where the bot is present, type `/hello`.
+1. You should see: `Hello <your Discord username here>!`
+![Bot response in Discord](../../media/bot-hello.png)
+
+{{< alert theme="info" >}}
+<img src="../../media/NF_mascot.jpg" alt="Nuvi mascot" width="70" style="float:right;margin:0 0 6px 10px;" />
+<strong>Nuvi tip:</strong> If `/hello` doesn’t autocomplete right away, wait up to 60 seconds or restart the bot—Discord is syncing commands.
+{{< /alert >}}
+
```

Comment:
Add additional challenges:

1. Update Hello message to add additional text like "Hello, username! Have a nice day!"
2. Create a new command like /smile or something to ensure students understand the command